File: stack.py
class Stack:
    def __init__(self):
        self.stc = []
        self.top = -1

    def push(self,val):
        self.top +=1  #top keep moving with insert
        self.stc.insert(self.top , val)
        
    def pop(self):
        try:
            temp = self.stc[self.top]
            print(temp)
            self.top -= 1 #top always remember pos 
        except Exception as e:
            print(e)

    def printstc(self):
        print(self.stc[:self.top+1]) #print till the top-1 pos   

stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)
stack.push(4)
stack.pop()
stack.pop()
stack.push("new " + str(3))
stack.push("new " + str(4))

stack.printstc()

File: stack.js
function stc(){    
    this.arr = []
    this.top = 0
    this.push = push
    this.pop = pop
    this.peek = peek
    this.print = print
}

function push(data){
    this.arr[this.top] = data
    this.top+=1
}

function pop(data){
    let temp = this.arr[this.top-1]
    this.top--
    console.log(temp)
}

function peek(data){
    console.log(this.arr[this.top-1])
}


function print(){
    for(let i=0; i<this.top; i++){
      console.log(this.arr[i])
}}

let obj = new stc()
obj.push("sam")
obj.push("sharan")
obj.peek()
obj.print()
obj.pop()
obj.print()
